plugins/net_plugin/net_plugin.cpp

```diff
@@ -287,6 +291,8 @@ namespace eosio {
       eosio::chain::named_thread_pool       thread_pool{ "net" };
 
 
+      unique_ptr<boost::asio::deadline_timer> accept_error_timer;
```

Comment:
Consider modifying to the new pattern in #632 that doesn't need this is be a a unique_ptr

Reply:
I'd have to move the initialization of `net_plugin_impl` into its constructor (right now it has the default constructor and `deadline_timer` doesn't default construct. It would be a good change, but beyond the scope of this issue. I'd be totally game to cleanup `net_plugin.cpp` but I don't think it is on the table for now.

Reply:
Admittedly I didn't look too closely, what's the problem with
```c++
eosio::chain::named_thread_pool       thread_pool{ "net" };
boost::asio::deadline_timer accept_error_timer{thread_pool.get_executor();};
```

Reply:
Right now these are initialized in `net_plugin::plugin_startup()` while `net_plugin_impl` is constructed in `net_plugin::net_plugin()`. So default-constructing these in the class would change when these are constructed. And `net_plugin::set_program_options()` can change `my->thread_pool_size` which is used in the thread pool constructor. 
So I'm not saying that this cannot be changed, just that it is beyond the original issue I think, and if we change this we probably want to do even more of a cleanup for consistency.

Reply:
Pretty sure it's just replacing
```c++
unique_ptr<boost::asio::deadline_timer> accept_error_timer;
```
with
```c++
boost::asio::deadline_timer accept_error_timer{thread_pool.get_executor()};
```

but it's your call

Reply:
You are right of course, I had not realized that the `named_thread_pool` created its `ioc`  in the constructor. My bad, and thanks for insisting! I will make the change.